Source/Charts/Data/Implementations/Standard/BarChartDataEntry.swift

```diff
@@ -156,38 +138,34 @@ open class BarChartDataEntry: ChartDataEntry
     /// - Returns:
     @objc open func calcRanges()
     {
-        let values = yValues
-        if values?.isEmpty != false
-        {
+        guard let values = yValues, !values.isEmpty else {
             return
         }
-        
+
         if _ranges == nil
         {
             _ranges = [Range]()
         }
         else
         {
-            _ranges?.removeAll()
+            _ranges!.removeAll()
```

Comment:
why we bother to force unwrap here? performance gain over `?`?

Reply:
Yes. `!` removes the runtime check for `nil`